contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -730,11 +730,9 @@ passed). The value of this field will be combined with a unique suffix on the
 server if the Name field has not been provided. The provided value must be valid
 within the rules for Name, and may be truncated by the length of the suffix
 required to make the value unique on the server. If this field is specified, and
-Name is not present, the server will NOT return a 409 if the generated name
-exists - instead, it will either return 201 Created or 504 with Reason
-`ServerTimeout` indicating a unique name could not be found in the time
-allotted, and the client should retry (optionally after the time indicated in
-the Retry-After header).
+Name is not present, the server will return a 409 with Reason `AlreadyExists` if
+the generated name exists, and the client should retry (optionally after the time
+indicated in the Retry-After header).
```

Comment:
```suggestion
the generated name exists, and the client should retry (after waiting at least the amount of time
indicated in the Retry-After header, if it is present).
```

(I was debating about making this comment or not)

Reply:
I wonder what is the effect of Retry-After header here? I see it in some other errors like server timeout or too many request, which makes sense and understandable. Is there any differences between a client chooses to retry with vs without respecting the header?

Reply:
The purpose is to keep the server from being overwhelmed by clients retrying in a hot loop, should the condition take some time to resolve.

Reply:
I'm OK to merge this with @lavalamp changes